main.cpp

```diff
@@ -0,0 +1,35 @@
+#include <app_tracing.h>
+
+/**
+ * To run:
+ * clear && g++ -I. -D__USE_XOPEN2K8 main.cpp build/opentracing/libopentracing.a -o main && ./main
+ */
+int main()
+{
+    // Setup
+    ZipkinV1 t1;
+    ZipkinV2 t2;
+
+    MyHotTracer imp(t1, t2);
+
+    GlobalTracer::install(&imp);
+
+
+    {
+        // Application code
+        auto tracer = Cpp11GlobalTracer::instance();
+
+        auto span = tracer.start("hi");
+
+        std::vector<opentracing::TextMapPair> pairs;
```

Comment:
Nice that this compiles but it likely means we have a bit too much freedom. I guess here's what's most of the discussion about. I guess both `ZipkinV1` and `ZipkinV2` should support something that serializes to text and http headers. If there's a proprietary IPC extension that `ZipkinV1` supported, I guess that `ZipkinV2` should support it as well to be backwards compatible, so no harm. If instead we had `ZipkinV1` and `Jaeger`, then I can see that `Jaeger` might not support the proprietary IPC extension (out of the box). It would be nice to be able to `#include` a single header file (with potentially infinite uglyness) and extend your `Jaeger` tracer to support the proprietary IPC. If this isn't possible, I guess the next alternative is to git-clone `Jaeger` and add support to it. In any case, for as long as Jaeger` doesn't support your custom IPC, the custom IPC infrastructure should be able to pretend it's a standard "HTTP" or "TEXT" type and it should work out of the box with potentially some inferior performance.